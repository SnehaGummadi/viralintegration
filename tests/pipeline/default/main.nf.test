nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.trace.tasks().size(),

                path("$outputDir/summary/test_T1.vif.prelim.refined.tsv"),
                // FIXME md5sum fails on CI
                // path("$outputDir/summary/test_T1.vif.refined.distilled.tsv"),
                // path("$outputDir/summary/test_T1.vif.refined.tsv"),
                // path("$outputDir/summary/test_T1.vif.refined.wRefGeneAnnots.tsv"),

                path("$outputDir/virus/novirus_T1.VirusDetect.igvjs.bam"),
                path("$outputDir/virus/novirus_T1.VirusDetect.sorted.igvjs.bam"),
                path("$outputDir/virus/redherring_T1.VirusDetect.igvjs.bam"),
                path("$outputDir/virus/redherring_T1.VirusDetect.sorted.igvjs.bam"),
                path("$outputDir/virus/test_T1.VirusDetect.igvjs.bam"),
                path("$outputDir/virus/test_T1.VirusDetect.sorted.igvjs.bam"),
                path("$outputDir/virus/test_T1.VirusDetect.virus_read_counts_summary.tsv"),
            ).match()

            assert new File("$outputDir/summary/test_T1.vif.refined.distilled.tsv").exists()
            assert new File("$outputDir/summary/test_T1.vif.refined.tsv").exists()
            assert new File("$outputDir/summary/test_T1.vif.refined.wRefGeneAnnots.tsv").exists()

            assert new File("$outputDir/summary/test_T1.vif.genome_plot.png").exists()
            assert new File("$outputDir/summary/test_T1.vif.html").exists()

            assert new File("$outputDir/virus/novirus_T1.VirusDetect.init.genome_plot.png").exists()
            assert new File("$outputDir/virus/novirus_T1.VirusDetect.virus_read_counts.png").exists()
            assert new File("$outputDir/virus/novirus_T1.VirusDetect.virus_read_counts_log.png").exists()
            assert new File("$outputDir/virus/redherring_T1.VirusDetect.init.genome_plot.png").exists()
            assert new File("$outputDir/virus/redherring_T1.VirusDetect.virus_read_counts.png").exists()
            assert new File("$outputDir/virus/redherring_T1.VirusDetect.virus_read_counts_log.png").exists()
            assert new File("$outputDir/virus/test_T1.VirusDetect.igvjs.html").exists()
            assert new File("$outputDir/virus/test_T1.VirusDetect.init.genome_plot.png").exists()
            assert new File("$outputDir/virus/test_T1.VirusDetect.virus_read_counts.png").exists()
            assert new File("$outputDir/virus/test_T1.VirusDetect.virus_read_counts_log.png").exists()
        }

    }

}
