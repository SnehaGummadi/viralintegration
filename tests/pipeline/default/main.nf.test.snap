{
    "Should run without failures": {
        "content": [
            45,
            "test_T1.vif.prelim.refined.tsv:md5,17b26b7a86d61206154ca3e2df02fc48",
            "test_T1.vif.refined.distilled.tsv:md5,ef63b5d294d3bbb93c8a836ac5301e3c",
            "test_T1.vif.refined.tsv:md5,4bac42d8c01d4c89850e7092854bcc3e",
            "test_T1.vif.refined.wRefGeneAnnots.tsv:md5,f6ce7c173e3706828557edd17c6fd33d",
            "novirus_T1.VirusDetect.igvjs.bam:md5,1416469ad13277d81204ff8332084ced",
            "novirus_T1.VirusDetect.sorted.igvjs.bam:md5,597eca709113196eae603bfe3e9dbe4e",
            "redherring_T1.VirusDetect.igvjs.bam:md5,190447f1542c0546b3c71b6957ade958",
            "redherring_T1.VirusDetect.sorted.igvjs.bam:md5,630a6554a6321bccc29dc5629961fb24",
            "test_T1.VirusDetect.igvjs.bam:md5,d72840f4fd77a1bcafc98191c4d9fbc7",
            "test_T1.VirusDetect.sorted.igvjs.bam:md5,399478ce0c0b561498c18ff8b77d91c4",
            "test_T1.VirusDetect.virus_read_counts_summary.tsv:md5,c051eb1a25e4741c43274589f75b80b6"
        ],
        "timestamp": "2022-12-11T17:28:21+0000"
    }
}